advisor_engine/foreman.py

```diff
@@ -30,7 +30,7 @@ def get_hosts_details(system_ids, batch_size = 1000):
         try:
             results.extend(api.resource_action('advisor_engine', 'host_details', params={"host_uuids": batch}))
         except Exception as e:
-            logger.error("An error occurred:", e)
+            logger.error(f"An error occurred: {e}")
```

Comment:
Isn't this a better way? https://docs.python.org/3/library/logging.html#logging.exception says it adds the exception info to the logging message and it actually tells you what it was trying to do
```suggestion
            logger.exception("An error occurred while getting host details")
```

Reply:
TIL

Quite possible, yes.

@brantleyr opinions?

Reply:
I'll note that I haven't tried it myself so perhaps the output is still useless, but I imagine services like Sentry are a lot more useful if you use it. At  least https://docs.sentry.io/platforms/python/integrations/logging/#verify says you'll get a stack trace.

Reply:
I am fine with either. Whatever you think is cleaner.

Reply:
Probably `logger.exception`

Reply:
I've tested the original fix by @evgeni and it got rid of the nasty traceback I was seeing. I'll be able to test other options next week :)

Reply:
The error being:
`<ERROR> - 2025-02-14 18:45:41,181 - An error occurred: Error while performing upload_hits on advisor_engine: 500 Server Error: Internal Server Error for url: https://satellite.red.ddns.perf.redhat.com/api/advisor_engine/upload_hits - {'message': 'PG::TRDeadlockDetected: ERROR:  deadlock detected\nDETAIL:  Process 981603 waits for ShareLock on transaction 7589113; blocked by process 1064519.\nProcess 1064519 waits for ShareLock on transaction 7589024; blocked by process 981603.\nHINT:  See server log for query details.\nCONTEXT:  while locking tuple (14,5) in relation "insights_rules"\n'}`